pkg/networkdevice/metadata/payload_utils.go

```diff
@@ -88,12 +92,88 @@ func appendToPayloads(namespace string, subnet string, collectTime time.Time, ba
 
 // DeviceOIDFromPDU packages a gosnmp PDU as a DeviceOID
 func DeviceOIDFromPDU(deviceID string, snmpPDU *gosnmp.SnmpPDU) (*DeviceOID, error) {
-	pdu, err := gosnmplib.PDUFromSNMP(snmpPDU)
+	oid := strings.TrimLeft(snmpPDU.Name, ".")
+	stringType, stringValue, base64Value, err := GetTypeAndValueFromPDU(snmpPDU)
 	if err != nil {
 		return nil, err
 	}
 	return &DeviceOID{
-		DeviceID: deviceID,
-		PDU:      pdu,
+		DeviceID:      deviceID,
+		Oid:           oid,
+		Type:          stringType,
+		ValueAsString: stringValue,
+		ValueAsBase64: base64Value,
 	}, nil
 }
+
+// GetTypeAndValueFromPDU returns the type as a string, the value as a string and the value encoded in base64 and an error if any
+func GetTypeAndValueFromPDU(pduVariable *gosnmp.SnmpPDU) (string, string, string, error) {
```

Comment:
Part of the goal of the `gosnmplib.PDU` type was so that if we need to write more complicated code to manage SNMP pdus in the future we would have one standard wrapper around `gosnmp.SnmpPDU` that serializes nicely and handles types properly. It also uses `gosnmplib.GetValueFromPDU`, which you're duplicating some logic from and missing some corner cases (e.g. it strips the leading '.' from OIDs).

We also don't need to handle cases like NoSuchObject or EndOfMibView, which shouldn't be coming from our gosnmp usage in the first place (and there's nothing we can do with them on the backend side if they do).

I think you should change this in your choice of two ways:
1. Replace `PDU.Value` with `PDU.ValueAsString` and `PDU.ValueAsBase64`, and putting the logic that checks `IsStringPrintable` in `gosnmplib.PDU.SetValue` **OR**
2. Use `GetValueFromPDU` here, and then keep the string conversions that you have here, and delete the `pdu` type from `gosnmplib` entirely (it's currently being used only by this file).